src/transaction/duck_transaction_manager.cpp

```diff
@@ -273,7 +273,9 @@ ErrorData DuckTransactionManager::CommitTransaction(ClientContext &context, Tran
 
 		auto &profiler = *context.client_data->profiler;
 		profiler.StartTimer(MetricsType::COMMIT_WRITE_WAL_LATENCY);
-		error = transaction.WriteToWAL(db, commit_state);
+		if (!DBConfig::GetSetting<NoWalModeSetting>(db.GetDatabase())) {
+			error = transaction.WriteToWAL(db, commit_state);
```

Comment:
I think this is actually the only change that is required - the other changes should not be necessary. The actual on-disk WAL is lazily created while writing so if we never write the WAL will not be created on-disk. The `WriteAheadLog` object can still exist in C++ even when this mode is enabled.